trieste/utils/__init__.py

```diff
@@ -14,7 +14,7 @@
 """ This package contains library utilities. """
 from typing import Callable, Mapping, TypeVar
 
-from .misc import DEFAULTS, Err, Ok, Result, T_co, jit, shapes_equal, to_numpy
+from .misc import DEFAULTS, Err, Ok, Result, ResultType, jit, shapes_equal, to_numpy
 
 K = TypeVar("K")
```

Comment:
a small digression: why is map_values defined in __init__ and not in e.g. misc.py?

Reply:
i meant to put everything in \_\_init\_\_.py but never got round to it. Given there's nothing else in `utils`, it now may as well be in a single utils.py file (no package)

Reply:
No idea! I'll move it.

Reply:
Ah ok. I'll migrate it in a separate PR.